src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let newStr = ``;
+
+    for (let i = 0; i < this.length; i++) {
+      let oneWord = this[i];
```

Comment:
This variable is redundant. Just use this[i] and don't overload your code with unnecessary variables.